test/app/helpers/twitter_intent.rb

```diff
@@ -0,0 +1,45 @@
+require 'active_support/core_ext/hash/slice'
+require 'active_support/core_ext/hash/conversions'
+require_relative '../../test_helper'
+require_relative '../../../app/helpers/twitter_intent'
+require 'rack'
+
+class TwitterIntentTest < Minitest::Test
+  TWITTER_URL = 'https://twitter.com/intent/tweet?'.freeze
+
+  def test_no_params
+    assert_equal TWITTER_URL, helper.twitter_share
+  end
+
+  def test_using_text
+    text = 'A #hashtag for @you http://exercism.io'
+    expected_uri =
+      "#{TWITTER_URL}%s" %
+      'text=A+%23hashtag+for+%40you+http%3A%2F%2Fexercism.io'
+
+    assert_equal expected_uri, helper.twitter_share(text: text)
+  end
+
+  def test_using_text_and_url
+    text = 'A #hashtag for @you'
+    url = 'http://exercism.io'
+    expected_uri =
+      "#{TWITTER_URL}%s" %
+      'text=A+%23hashtag+for+%40you&url=http%3A%2F%2Fexercism.io'
+
+    assert_equal expected_uri,
+                 helper.twitter_share(text: text, url: url)
+  end
+
+  def test_other_params
```

Comment:
`test_other_params_are_ignored` will be gone if we use optional keyword arguments.